R/plot_timeSeries.R

```diff
@@ -0,0 +1,378 @@
+#' If `FALSE` the single-model ensembles are plotted individually. 
+#' @param simplify logical. Simplify the ensemble range and mean using a smoothing spline. 
+#' @param refline logical. Plot the 1961-1990 reference period mean for the selected variable
+#' and extend this line to the year 2100 as a visual reference. 
+#' @param refline.obs logical. Plot the 1961-1990 reference period mean for the observational data. 
+#' This should be the same as the reference line for the GCM time series.  
+#' @param label.endyear logical. Add a label of the final year of the observational time series. 
+#' @param endlabel character. Add a label to the end of each simulated time series. Options 
+#' are "change", to indicate the change in year 2100 relative to the 1961-1990 baseline, or "model"
+#' to indicate the global climate model. 
+#' @param yearmarkers logical. Add white points to the observational time series as a visual aid. 
+#' @param yearlines logical. Add vertical lines on every fifth year as a visual reference
+#' @param legend_pos character. Position of the legend. Viable options are `c("bottomright",
+#'   "bottomleft", "topleft", and "topright")`.
+#'
+#' @return NULL. Draws a plot in the active graphics device.
+#'
+#' @importFrom graphics axis legend lines par points text title
+#'
+#' @examples
+#' # data frame of arbitrary points
+#' my_points <- data.frame(lon = c(-127.7300,-127.7500), lat = c(55.34114, 55.25), elev = c(711, 500), id = 1:2)
+#'
+#' # generate the input data
+#' my_data <- plot_timeSeries_input(my_points)
+#'
+#' # use the input to create a plot
+#' plot_timeSeries(my_data)
+#'
+#' # compare mean daily minimum and maximum temperatures
+#' plot_timeSeries(my_data, variable1 = "Tmin_sm", variable2 = "Tmax_sm")
+#'
+#' # compare summer and winter temperatures
+#' plot_timeSeries(my_data, variable1 = "Tmax_sm", variable2 = "Tmax_wt")
+#'
+#' # compare global climate models
+#' plot_timeSeries(my_data, variable1 = "Tmin_sm", gcms.ts = list_gcm()[c(13,7)], ssp = list_ssp()[2], compile = F, simplify = F, endlabel = "model", mar=c(3,3,0.1,6), showObserved = F)
+#'
+#' # export plot to a temporary directory
+#' figDir <- tempdir()
+#' png(
+#'   filename = file.path(figDir, "plot_test.png"), type = "cairo", units = "in",
+#'   width = 6, height = 5, pointsize = 10, res = 300
+#' )
+#' plot_timeSeries(my_points, variable1 = "Tmin_sm")
+#' dev.off()
+#'
+#' @export
+
+plot_timeSeries <- function(
+    X,
+    variable1 = "Tmin_sm",
+    variable2 = NULL,
+    showObserved = T,
+    historic_ts_dataset = "climate_na",
+    gcms.ts = list_gcm()[c(1,4,5,7,10,11,12)], #TODO add GFDL once the data are in climr
+    ssps = list_ssp()[1:3],
+    showrange = T, 
+    yfit = T,
+    cex = 1,
+    mar=c(3,3,0.1,4), 
+    showmean = T,
+    compile = T,
+    simplify = T,
+    refline = T,
+    refline.obs = T,
+    label.endyear = F, 
+    endlabel = "change", 
+    yearmarkers = T,
+    yearlines = F,
```

Comment:
please replace all `T` and all `F` by `TRUE` and `FALSE`, here and everywhere  else
see https://www.r-bloggers.com/2012/12/r-tip-avoid-using-t-and-f-as-synonyms-for-true-and-false/